lib/run-tests/worker.js

```diff
@@ -0,0 +1,72 @@
+const path = require("path");
+const crypto = require("crypto");
+const lmdb = require("lmdb");
+
+const BabelAgent = require("./babel-agent");
+const transpile = require("./transpile");
+
+const cache = lmdb.open({
+  path: "cache.db",
+  compression: true,
+});
+
+const TEST_TIMEOUT = 60 * 1000; // 1 minute
+
+function timeout(file, waitMs = TEST_TIMEOUT) {
+  return new Promise((_resolve, reject) =>
+    setTimeout(
+      () => reject(new Error(`test ${file} timed out after ${waitMs} ms`)),
+      waitMs
+    )
+  );
+}
+
+let agent, testRoot;
+
+exports.setup = function (opts) {
+  agent = new BabelAgent(opts);
+  testRoot = opts.testRoot;
+};
+
+exports.runTest = async function (test) {
+  let { attrs, contents, file } = test;
+  const isModule = attrs.flags.module;
+
+  try {
+    contents = transpile(contents, attrs.features, isModule);
+  } catch (error) {
+    return { result: "parser error", error };
+  }
+
+  const hash = crypto.createHash("sha1").update(contents).digest("hex");
+
+  const cached = cache.get(hash);
+  if (cached) return cached;
```

Comment:
What is this needed for? :thinking: Are there duplicated tests?

Reply:
Is it possible to have all properties as cache keys?

Reply:
Whops, I merged before seeing this.

Potentially yes, but I'm still not sure that it will work. Some tests have dependencies on other files, and we should invalidate the cache also if one of the dependencies changes.

Reply:
OK, let's look at the new execution time first. If it is still slow I will try to use the cache again.